site/app/libraries/database/DatabaseQueries.php

```diff
@@ -3653,6 +3661,11 @@ private function updateGradedComponents(TaGradedGradeable $ta_graded_gradeable)
                     $deleted_marks = array_diff($component_grade->getDbMarkIds() ?? [], $component_grade->getMarkIds());
                     $this->deleteGradedComponentMarks($component_grade, $deleted_marks);
                     $this->createGradedComponentMarks($component_grade, $new_marks);
+
+                    //change the component to be unverified after changing a mark
+                    $component_grade->setVerifier(null);
+                    $component_grade->setVerifyTime(null);
+                    $this->updateGradedComponent($component_grade);
```

Comment:
We should avoid this kind of logic in the database saving functions. Maybe put it in `saveGradedComponent`?